logger/logger.go

```diff
@@ -163,7 +163,7 @@ func (l *Log) debugEnabled() bool {
 // Debug only prints if SNAPD_DEBUG is set
 func (l *Log) Debug(msg string) {
 	if l.debugEnabled() {
-		l.NoGuardDebug(msg)
+		l.log.Output(3, "DEBUG: "+msg)
```

Comment:
This could use a comment explaining what is the issue. I suspect it's about call stack frame depth and how we figure out the caller file name, but whoever looks at it later will be left wondering.

Reply:
Yeah, that's true. I will update the code and git history to explain the problem and re-request a review.